### PR TITLE
[dcp-481] Job for populating dcp versions

### PR DESCRIPTION
**Pull request by aaclan-ebi**
ebi-ait/dcp-ingest-central#481

Moved https://github.com/ebi-ait/hca-ebi-dev-team/pull/453 to here

### CODE REVIEW

File: cron-jobs/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Jobs
+This is a directory for CRON jobs ran inside the cluster
+
+## Creating jobs
+1. Copy one of the helm carts in this directory (e.g. `update-project-catalogue`) and rename to your needs
```
Comment (amnonkhen) [inline cron-jobs/README.md:5]: I would add an actual link to the example chart.

File: cron-jobs/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+2. Follow the format of the helm chart and edit as needed
+3. Docker container (the one that is deployed to k8s and ran as a CRON job) is located in the `docker-app` subdir
+4. Ensure a file called `deploy.sh` is present in the root of the helm chart. This is needed for the `deploy-all.sh` script
+
+## Deploying all jobs
+1. `source config/environment_<env>` (from ingest-kube-deployment root)
+2. `./deploy-all.sh` (from this folder)
```
Comment (amnonkhen) [inline cron-jobs/README.md:12]:
```suggestion

	  ```bash
	  # run from ingest-kube-deployment root 
	  source config/environment_<env>
	  # run from cron-jobs root
	  ./deploy-all.sh
	  ```

```

File: jobs/dcp1-version/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v2
+name: dcp1-version
+description: A Helm chart for Kubernetes
```
Comment (amnonkhen) [inline jobs/dcp1-version/Chart.yaml:3]: Where does this description appear? should it link to the readme?

File: jobs/dcp1-version/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Populating firstDcpVersion and dcpVersion of DCP1 metadata
+
+The script `populate_dcp_version.py` fixes the missing `dcpVersion` values in the metadata documents in the Ingest database. It reads `dcp1-metadata.txt.tar.gz`  which is the compressed file that contains the list of all the metadata json files in the DCP1 terra bucket `gs://broad-dsp-monster-hca-prod-ucsc-storage/prod/no-analysis/metadata`.
+
+Please see issue: https://github.com/ebi-ait/dcp-ingest-central/issues/481
```
Comment (amnonkhen) [inline jobs/dcp1-version/README.md:5]: 👍

File: jobs/dcp1-version/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+## How to build docker image and push to quay.io?
+
+```
+$ docker build . -t quay.io/ebi-ait/ingest-kube-deployment:dcp1-version_20211102.1
+$ docker push quay.io/ebi-ait/ingest-kube-deployment:dcp1-version_20211102.1
+```
```
Comment (amnonkhen) [inline jobs/dcp1-version/README.md:12]:
```suggestion
```bash
docker build . -t quay.io/ebi-ait/ingest-kube-deployment:dcp1-version_20211102.1
docker push quay.io/ebi-ait/ingest-kube-deployment:dcp1-version_20211102.1
```
```

File: jobs/dcp1-version/README.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+## How to run inside ingest cluster?
+Source environment config and set correct environment context 
+```
+$ source config/environment-dev
+$ kubectx ingest-eks-dev
+```
```
Comment (amnonkhen) [inline jobs/dcp1-version/README.md:19]: I think the `$` make it more difficult to copy and past the code examples.
```suggestion
```bash
source config/environment-dev
kubectx ingest-eks-dev
```
```

File: jobs/dcp1-version/README.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+```
+# find the pod name first
+$ kubectl get pods | grep dcp1-version
+dcp1-version-lgmj2                                      1/1     Running     0          22s
+
+# follow logs
+$ kubectl logs -f dcp1-version-lgmj2
+<logs>
+```
```
Comment (amnonkhen) [inline jobs/dcp1-version/README.md:36]:
```suggestion
```bash
# find the pod name first
kubectl get pods | grep dcp1-version
dcp1-version-lgmj2                                      1/1     Running     0          22s

# follow logs
kubectl logs -f dcp1-version-lgmj2
<logs>
```
```

File: jobs/dcp1-version/README.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+To delete the job
+
+```
```
Comment (amnonkhen) [inline jobs/dcp1-version/README.md:40]:
```suggestion
```bash
```

File: jobs/dcp1-version/docker-app/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,12 @@
+FROM quay.io/ebi-ait/ingest-base-images:python_3.7-slim
+
+COPY scripts/populate_dcp_version.py populate_dcp_version.py
+COPY scripts/requirements.txt requirements.txt
```
Comment (amnonkhen) [inline jobs/dcp1-version/docker-app/Dockerfile:4]: move this above the py file to optimise build time

File: jobs/dcp1-version/docker-app/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+COPY scripts/dcp1-metadata.txt.tar.gz dcp1-metadata.txt.tar.gz
+
+RUN tar -xzvf dcp1-metadata.txt.tar.gz
+
+ENV DCP1_GS_FILES_LIST=dcp1-metadata.txt
+
+RUN pip install -r requirements.txt
```
Comment (amnonkhen) [inline jobs/dcp1-version/docker-app/Dockerfile:11]: move this after the requirements.txt COPY and before the py file copy to optimise build time by using the docker cache

File: jobs/dcp1-version/docker-app/scripts/populate_dcp_version.py [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+import uuid
+from datetime import datetime
+
+from bson import JAVA_LEGACY, CodecOptions
+from pymongo import MongoClient
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+DCP1_GS_FILES_LIST = os.environ.get('DCP1_GS_FILES_LIST')
```
Comment (amnonkhen) [inline jobs/dcp1-version/docker-app/scripts/populate_dcp_version.py:12]: If there is no default value, there should probably be validation that the value is populated.

**Review by amnonkhen — CHANGES_REQUESTED** [review]

**Review by amnonkhen — COMMENTED** [review]
overall looks good, minor dockerfile/python/doc comments.

File: jobs/dcp1-version/Chart.yaml (hunk 3 above)
Comment (jacobwindsor) [reply jobs/dcp1-version/Chart.yaml:3]: I don't think it's necessary. These fields are more for charts that are being published. Similiar to description in package.json

File: jobs/dcp1-version/docker-app/Dockerfile (hunk 9 above)
Comment (aaclan-ebi) [reply jobs/dcp1-version/docker-app/Dockerfile:3]: Thanks!

File: jobs/dcp1-version/docker-app/Dockerfile (hunk 10 above)
Comment (aaclan-ebi) [reply jobs/dcp1-version/docker-app/Dockerfile]: Applied.

File: jobs/dcp1-version/docker-app/scripts/populate_dcp_version.py (hunk 11 above)
Comment (aaclan-ebi) [reply jobs/dcp1-version/docker-app/scripts/populate_dcp_version.py:12]: Good point, thanks!

**Review by jacobwindsor — APPROVED** [review]

**Review by ke4 — APPROVED** [review]
LGTM 👍

**Review by amnonkhen — APPROVED** [review]

**Pull request merged**